static/src/js/util/map/getQueryFromShapefileFeature.ts

```diff
@@ -2,10 +2,27 @@ import { Feature } from 'ol'
 import { Point } from 'ol/geom'
 
 import spatialTypes from '../../constants/spatialTypes'
-import { SpatialQueryType } from '../../types/sharedTypes'
+import { Spatial, SpatialQueryType } from '../../types/sharedTypes'
+
+// Coordinate types for spatial types
+type Coordinate = [number, number] | [number, number, number]
+type LineString = Coordinate[]
+type Polygon = Coordinate[][]
+type MultiPolygon = Coordinate[][][]
+type CoordinateInput = Coordinate | LineString | Polygon | MultiPolygon
+
+// Remove altitude from coordinate arrays recursively
+const removeAltitudeFromCoordinates = (coordinates: CoordinateInput): CoordinateInput => {
```

Comment:
It's weird to use `CoordinateInput` as the name for the input and output of this function. Maybe `CoordinateGroup` or `Coordinates` might work better?